From f32f88366eac326539e189f36400adffbc744676 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 5 Feb 2020 15:07:57 +0100
Subject: [PATCH] PONRTSYS-6642: Fix compile error when UART is disabled

This adds some ifdefs to deactivate the code which calls the TEP UART
driver. Without this changes the bootcore image does not build.

This fixes the following link errors:
mips-openwrt-linux-musl-ld: arch/mips/built-in.o: in function `uart_interrupt':
arch/mips/lantiq/grx500_bootcore/irq.c:146: undefined reference to `grx500_bootcore_uart_tx_isr'
mips-openwrt-linux-musl-ld: arch/mips/lantiq/grx500_bootcore/irq.c:151: undefined reference to `grx500_bootcore_uart_rx_isr'
mips-openwrt-linux-musl-ld: arch/mips/built-in.o: in function `early_lntq_write':
arch/mips/lantiq/grx500_bootcore/early_printk.c:201: undefined reference to `uart_console_write'
---
 arch/mips/lantiq/grx500_bootcore/Makefile | 3 ++-
 arch/mips/lantiq/grx500_bootcore/irq.c    | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/arch/mips/lantiq/grx500_bootcore/Makefile b/arch/mips/lantiq/grx500_bootcore/Makefile
index 662ccde23070..afa436453d8d 100644
--- a/arch/mips/lantiq/grx500_bootcore/Makefile
+++ b/arch/mips/lantiq/grx500_bootcore/Makefile
@@ -1,4 +1,5 @@
-obj-y := prom.o clk.o sysctrl.o early_printk.o irq.o
+obj-y := prom.o clk.o sysctrl.o irq.o
+obj-$(CONFIG_SERIAL_GRX500_BOOTCORE_CONSOLE) += early_printk.o
 obj-$(CONFIG_SOC_GRX500_BOOTCORE) += ngi.o
 obj-$(CONFIG_SOC_PRX300_BOOTCORE) += ngi_prx300.o
 #obj-y := prom.o clk.o irq.o sysctrl.o early_printk.o ngi.o timer.o
diff --git a/arch/mips/lantiq/grx500_bootcore/irq.c b/arch/mips/lantiq/grx500_bootcore/irq.c
index 60b03ad78d89..3f0d9cfd7956 100644
--- a/arch/mips/lantiq/grx500_bootcore/irq.c
+++ b/arch/mips/lantiq/grx500_bootcore/irq.c
@@ -141,6 +141,7 @@ inline void grx500_bootcore_ack_irq(unsigned int irq_nr)
         UartStatus = MT_RdReg(MT_LOCAL_MIPS_BASE_ADDRESS, REG_UART_INT);
     printk("exit: UNMAPPED_IRQS_TO_MAPPER = %x, MAPPED_IRQS_FROM_MAPPER = %x, stat b=%x ", temp1, temp2, UartStatus);
 #endif
+#ifdef CONFIG_SERIAL_GRX500_BOOTCORE_CONSOLE
         // TX Part
         if (UartStatus && REG_UART_WRITE_FIFO_WATER_LEVEL_IRQ_MASK)
             grx500_bootcore_uart_tx_isr(dev_id); /*needed???*/
@@ -149,6 +150,7 @@ inline void grx500_bootcore_ack_irq(unsigned int irq_nr)
         if ((UartStatus && REG_UART_READ_FIFO_WATER_LEVEL_IRQ_MASK) ||
             (UartStatus && REG_UART_IDLE_IRQ_MASK))
         grx500_bootcore_uart_rx_isr(dev_id);
+#endif /* CONFIG_SERIAL_GRX500_BOOTCORE_CONSOLE */
          
     return IRQ_HANDLED;
 }
